web_src/js/features/repo-settings-branches.ts

```diff
@@ -0,0 +1,30 @@
+import {createSortable} from '../modules/sortable.ts';
+import {POST} from '../modules/fetch.ts';
+
+export function initRepoBranchesSettings() {
+  const protectedBranchesList = document.querySelector(
+    '#protected-branches-list',
+  );
+  if (!protectedBranchesList) return;
+
+  createSortable(protectedBranchesList, {
+    handle: '.drag-handle',
+    animation: 150,
+    onEnd: async () => { // eslint-disable-line @typescript-eslint/no-misused-promises
+      const newOrder = Array.from(protectedBranchesList.children, (item) => {
+        const id = item.getAttribute('data-id');
+        return id ? parseInt(id) : NaN;
+      }).filter((id) => !Number.isNaN(id));
+
+      try {
+        await POST(protectedBranchesList.getAttribute('data-priority-url'), {
+          data: {
+            ids: newOrder,
+          },
+        });
+      } catch (err) {
+        console.error('Failed to update branch protection rule priority:', err);
```

Comment:
It needs to report to the UI, otherwise users won't know that the updating failed.

Reply:
![image](https://github.com/user-attachments/assets/2636dbc8-92ef-4d3e-9bba-d4c4bd3e8efa)

Reply:
(ffff47d6355caa4fe39a87dd4688db4a8d31b014)